modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -207,7 +209,8 @@ public void clusterChanged(ClusterChangedEvent event) {
             }
         }
 
-        if (event.metadataChanged() && event.changedCustomMetadataSet().contains(IngestMetadata.TYPE)) {
+        if (event.metadataChanged()
+            && (event.indicesCreated().isEmpty() == false || event.changedCustomMetadataSet().contains(IngestMetadata.TYPE))) {
```

Comment:
ℹ️ Now we need to check if an index is created with the pipeline set.
TODO: what about update?